src/Aspire.Hosting.Azure/BicepGenerationCallbackAnnotation.cs

```diff
@@ -0,0 +1,11 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+
+namespace Aspire.Hosting.Azure;
+
+internal class BicepGenerationCallbackAnnotation(Action<BicepGenerationContext> callback) : IResourceAnnotation
```

Comment:
This annotation holds a callback for each resource that is being converted into an `azure.bicep.v0` resource in the manifest. We use a single manifest writer callback and it sets up the bicep generation context and calls the callback attached to this annotation. The callback may point to a default implementation or an override with the developers customizations.

The intent is that there is one bicep generation callback for each resource that generates a Bicep file. For parent/child relationships, the will probably be responsible for generating any Bicep for the child resources via the Azure.Provisioning API surface (as is shown in this PR).